BeerDispancer/BeerDispencer.Infrastructure/Payment/PaymentProcessorJob.cs

```diff
@@ -0,0 +1,88 @@
+﻿using BeerDispencer.Infrastructure.Persistence.Abstractions;
+using BeerDispencer.Infrastructure.Persistence.Entities;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace BeerDispencer.Infrastructure.Payment
+{
+    public class PaymentProcessorJob: BackgroundService
+    {
+       
+        private readonly IServiceScopeFactory facrory;
+        private readonly ILogger<PaymentProcessorJob> _logger;
+
+        public PaymentProcessorJob(IServiceScopeFactory facrory , ILogger<PaymentProcessorJob> logger, IHostApplicationLifetime applicationLifetime)
```

Comment:
How can you be sure that PaymentProcessorJob won't be run in 2 separate services?